htlcswitch/mailbox.go

```diff
@@ -366,8 +392,15 @@ func (m *memoryMailBox) mailCourier(cType courierType) {
 		// doesn't make it into a commitment, then it'll be
 		// re-delivered once the link comes back online.
 		case pktCourier:
-			nextPkt = m.pktHead.Value.(*htlcPacket)
-			nextPktEl = m.pktHead
+			// Peek at the next item to deliver, prioritizing
+			// Settle/Fail packets over Adds.
+			if m.pktHead != nil {
+				nextPkt = m.pktHead.Value.(*htlcPacket)
```

Comment:
Cool, I wonder if there's any noticeable perf improvement over long links for stuff like benchmark throughput tests.